06_iito/index.tex

```diff
@@ -24,11 +27,38 @@ \section{Inter-Island Communication}
-\label{sec:IITO:gifting}  
+\subsection{Gifting}  
+\label{subsec:IITO:gifting}  
 
 If an island is struggling and requires some additional amount of resources to survive, it may ask the President for an allocation from the common pool but the President may reject this request. The struggling island would still be able to take resources from the common pool if rejected, but law-abiding islands would probably want to avoid such disobedience. To allow islands to still request an additional amount of resources from the other islands, the \textbf{gifting} action is an option. Islands may accept the gift requests if they wish to improve their standing with the requesting islands. They can also accept the gift requests of those whose survival is deemed to be essential for the future of the archipelago. When requesting, giving or receiving a gift, the island can specify a reason for this action, which gives islands more information about the gifting transaction. For example, a requesting island may want to specify that they will move to a critical state in the next round if they do not receive the gift, or an offering island may want to specify that the gift is meant as a reward for successful disaster forecasting.
 
-Islands are also able to offer gifts without a request being made, meaning they can reach out to other islands if they want to boost their popularity.
\ No newline at end of file
+Islands are also able to offer gifts without a request being made, meaning they can reach out to other islands if they want to boost their popularity.
+
+\section{Implementation}
+\label{sec:IITO:Implementation}
+
+\subsection{Server Client Infrastructure}
+\label{subsec:IITO:server_client_infrastructure}  
+
+All IITO sessions, such as Gift Giving or Common Pool Donations, follow the same client server interaction format. The server acts a middleman for all agent to agent communication, and prompts agents to make decisions or formulate messages during the session. At each stage in a session the server compiles a list of messages from the agents, following that sessions' protocols, and then uses that list as input for the next stage. While a bit abstract this is better explained through an example.
+
+\subsection{Gifting Session}
+\label{subsec:IITO:gifting_session} 
+
+There are four phases to the gifting sessions: Gift requests, Gift offers, Gift responses and Updating gift history. Each part is further explained below and Figure~\ref{fig:IITO:gifting_session_diagram} shows the interactions a single client will have with the server over the session.
+\begin{itemize}
+    \item \textbf{Gift Requests:} Collects the gift requests of all islands. These requests include an amount the client wants and who they wish to request from.
+    \item \textbf{Gift Offers:} Gives the clients any requests that are directed to them and then collects the gift offers from each client. These offers include an amount the client wished to gift and to whom the gift is being offered.
```

Comment:
make a full-sentence -- do NOT start with the verb directly. (looks bad)

clients ---> islands (terminology 'client' has a different meaning in infra context thoo.... but here, you mean islands)

suggested re-writing:
\item \textbf{Gift Offers}:  **It gives the islands**